Release v0.17.0 (#1134)
firecracker-microvm/firecracker: docs/getting-started.md

- **Linux 4.14+**

>   Firecracker currently supports physical Linux **x86_64** and **aarch64**

Reviewer: We should specify that aarch64 is not feature complete at it is considered alpha.

Author: Done, take a look.

the basic requirements to run Firecracker.</summary>

```bash

>     #!/bin/bash

Reviewer 2: I don't think there's need to include the shebang too, I don't expect readers to copy this snippet, save it in a file, then run it. I'd just paste it in a terminal.

[built from source](#building-from-source).

Next, you will need an uncompressed Linux kernel binary, and an ext4
file system image (to use as rootfs).

> 1. To run a `x86_64` guest you can download such resources from:

Reviewer 2: ```suggestion
1. To run an `x86_64` guest you can download such resources from:
```

- get the kernel and rootfs, if you don't have any available:

  ```bash
    #!/bin/bash

    arch=`uname -m`
    dest_kernel="hello-vmlinux.bin"
    dest_rootfs="hello-rootfs.ext4"

>     if [ $arch = "x86_64" ]; then

Reviewer 2: nit: other variables in the existing bash snippets are dereferenced with `${}` not just `$`

>             kernel="https://s3.amazonaws.com/spec.ccfc.min/img/hello/kernel/hello-vmlinux.bin"

Reviewer 2: Can you add a prefix variable for `https://s3.amazonaws.com/spec.ccfc.min/img`?

            rootfs="https://s3.amazonaws.com/spec.ccfc.min/img/hello/fsfiles/hello-rootfs.ext4"
    elif [ $arch = "aarch64" ]; then
            kernel="https://s3.amazonaws.com/spec.ccfc.min/img/aarch64/ubuntu_with_ssh/kernel/vmlinux.bin"
            rootfs="https://s3.amazonaws.com/spec.ccfc.min/img/aarch64/ubuntu_with_ssh/fsfiles/xenial.rootfs.ext4"
    else
            echo "Cannot run firecracker on $arch architecture!"
            exit 1
    fi

> 

Reviewer 2: nit: remove extra blank line

Within the Firecracker repository root directory:

> 1. on a **x86_64** machine:  ```tools/devtool build```

Reviewer 2: ```suggestion
1. on an **x86_64** machine:  ```tools/devtool build```
```

> 1. on an **aarch64** machine: ```tools/devtool build -- --target aarch64-unknown-linux-musl```

Reviewer: Why isn't musl the default on aarch64?

Author: So, cargo will use as default the thing you specify in the build section of the cargo/.config.
Ours looks like this.
```
[build]
target = "x86_64-unknown-linux-musl"
```
So, it will try to build with this target on aarch64. Unfortunately I have not found a way to adjust it based on architecture. I am open to suggestions.
I tried using:
```
[build.'cfg(target_arch = "x86_64")']
target = "x86_64-unknown-linux-musl"

[build.'cfg(target_arch = "aarch64")']
target = "aarch64-unknown-linux-musl"

```
but cargo will ignore them and start considering default target the gnu target even for x86_64 arch so it seems that it breaks even the x86_64 settings.
I am open to get help here.

Reviewer 3: `.cargo/config` doesn't seem to support having multiple targets based on architecture: https://github.com/rust-lang/cargo/issues/3913.

and, as such, some performance tests may fail when run on a regular desktop
machine. Specifically, don't be alarmed if you see
`tests/integration_tests/performance/test_process_startup_time.py` failing when

> not run on an EC2 .metal instance. You can avoid running performance tests with:

Reviewer 2: ```suggestion
not run on an EC2 .metal instance. You can skip performance tests with:
```
